Executable/Program.cs

```diff
@@ -47,7 +46,47 @@ internal static void Main(string[] args)
                     Environment.Exit(1);
                 }
 
-                if (Directory.GetFiles(directory, "*Subnautica*.exe", SearchOption.TopDirectoryOnly).Length <= 0)
+                string windowsDirectory = Path.Combine(Environment.CurrentDirectory, "../..");
+                string macDirectory = Path.Combine(Environment.CurrentDirectory, "../../../../..");
+
+                // Check if the device is running Windows OS
+                bool onWindows;
+                if (!Directory.Exists(windowsDirectory)) onWindows = false;
+                else
+                {
+                    try
+                    {
+                        // Try to get the Subnautica executable
+                        // This method throws a lot of exceptions
+                        onWindows = Directory.GetFiles(windowsDirectory, "Subnautica.exe", SearchOption.TopDirectoryOnly).Length > 0;
+                    }
+                    catch
+                    {
+                        // If an exception was thrown, the file probably isn't there
+                        onWindows = false;
+                    }
+                }
+
+                // Check if the device is running Mac OS
+                bool onMac;
+                if (!Directory.Exists(macDirectory)) onMac = false;
```

Comment:
> Is it possible for this mostly repeated code to be refactored into a single, generalized, and more maintainable method?

It may be similar, but it's not identical. I would prefer not to make a method with 3 arguments just tot avoid repeating a bit of code.